frontend/src/components/annotation/annotation_core.vue

```diff
@@ -3861,18 +3875,39 @@ mplate_has_keypoints_type: function (instance_template) {
     },
 
     insert_tag_type: function () {
-      this.push_instance_to_instance_list_and_buffer(
+      this.add_instance_to_file(
         this.current_instance,
         this.current_frame
       );
     },
+    set_frame_pending_save: function(value, frame_number){
+      if(frame_number == undefined){
+        return
+      }
+      if (this.instance_buffer_metadata[frame_number]) {
+        // We need to recreate object so that computed props get triggered
+        this.instance_buffer_metadata[frame_number].pending_save = value;
+
+      } else {
+        this.instance_buffer_metadata[frame_number] = {
+          pending_save: value
+        }
+      }
+      // Keep unsaved_frames list to enable/disable save button
+      if(value){
+        this.unsaved_frames.push(frame_number)
```

Comment:
This is excellent Pablo, it makes the video file abstraction stronger, a save saves the entire file across frames.